### PR TITLE
Remove testing hacks

### PR DESCRIPTION
**Pull request by SUPERCILEX**
This solution isn't great since it means we'll have to mock every `FirebaseApp` method we use, but it gives us absolute control over every Firebase API.

### CODE REVIEW

File: auth/src/test/java/com/firebase/ui/auth/testhelpers/TestHelper.java [hunk 1]
```diff
@@ -72,18 +74,42 @@ private static void spyContextAndResources() {
...
+        for (Field field : FirebaseApp.class.getDeclaredFields()) {
+            field.setAccessible(true);
+
+            Object o;
+            try {
+                o = field.get(null);
+            } catch (IllegalAccessException e) {
+                throw new IllegalStateException(e);
+            } catch (NullPointerException e) {
+                continue; // Instance field, move on
```
Comment (piotrek1543) [inline auth/src/test/java/com/firebase/ui/auth/testhelpers/TestHelper.java]: Seems like one hack is still here ;-)

Comment (SUPERCILEX) [reply auth/src/test/java/com/firebase/ui/auth/testhelpers/TestHelper.java]: Lol, yeah. The more accurate title would be `Remove testing hacks in production in favor of hacks in testing code`.

**Comment by SUPERCILEX** [comment]
@samtstern BTW, all 4.1 PRs are going to fail until #1329 is resolved.

**Comment by samtstern** [comment]
@SUPERCILEX are you sure this is better?  It still relies on reflection that's likely to break one day.  May not be any less of a hack than my half-assed dependency injection.

**Comment by SUPERCILEX** [comment]
Yeah, it's not great, but it moves the hacks to our tests instead of the production code. If you don't agree that that's better (I'm on the fence myself), I'll clean up your injection instead. 👍

**Comment by samtstern** [comment]
If you have an idea to clean up the injection I'd be all for that instead!

**Comment by SUPERCILEX** [comment]
@samtstern Voila! Yeah, you're right, this is _way_ better. Thanks! ❤️

**Comment by samtstern** [comment]
@SUPERCILEX nice!

**Pull request merged**